TheWitcher.Business/HeroService.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                if(quest.StartTime.Value < DateTime.Now)
+                {
+                    hero.HeroMoney += quest.Quest.Award;
+                    _heroInQuestRepository.Delete(quest.Id);
+                }
+            }
+            _heroRepository.Update(hero);
+        }
+
+        public bool TakeTheQuest(int heroId, int questId)
+        {
+            var hero = _heroRepository.GetItem(heroId);
+            var quest = _questRepository.GetItem(questId);
+            int heroPower = CountPowerOfHero(heroId);
+            if (heroPower > quest.Complexity)
+            {
+                if (hero.ReleaseDate < DateTime.Now || hero.ReleaseDate == null)
+                {
+                    ProcessCoefficient(hero, quest, heroPower);
+
+                    HeroInQuest heroInQuest = new HeroInQuest()
+                    {
+                        HeroId = heroId,
+                        QuestId = questId,
+                        StartTime = DateTime.Now,
+                    };
+                    _unitOfWork.BeginTransaction();
+                    try
+                    {
+                        _heroInQuestRepository.Create(heroInQuest);
+                        _heroRepository.Update(hero);
+                        _questRepository.Update(quest);
+                        _unitOfWork.EndTransaction();
+                    }
+                    catch(Exception e)
+                    {
+                        _unitOfWork.RollBack();
+                        return false;
+                    }
+                    return true;
+                }
+            }
+            return false;
+        }
+       
+        private bool ChangeHeroMoney(int heroId, int money, char sign)
+        {
+            try
+            {
+                var hero = _heroRepository.GetItem(heroId);
+                if (sign == '+')
+                    hero.HeroMoney += Convert.ToDecimal(money);
+                else
+                    hero.HeroMoney -= Convert.ToDecimal(money);
+                _heroRepository.Update(hero);
+            }
+            catch (Exception e)
+            {
+                return false;
+            }
+            return true;
+        }
+        public bool BuyClothes(int heroId, int clothesId)
+        {
+            var hero = _heroRepository.GetItem(heroId);
+            var cloth = _clothesRepository.GetItem(clothesId);
+            if(hero.HeroMoney> cloth.PriceOfBuy && hero.AvailableWeight > cloth.ClothesWeight.Value)
+            {
+                hero.HeroMoney -= cloth.PriceOfBuy;
+                hero.AvailableWeight -= cloth.ClothesWeight.Value;
+                HeroClothes heroClothes = new HeroClothes()
+                {
+                    HeroId = heroId,
+                    Heroes = hero,
+                    ClothesId = clothesId,
+                    Clothes = cloth,
+                    IsActive = false,
+                    PriceOfSell = cloth.PriceOfBuy / 3
```

Comment:
3 should be constant
false can be made a constant (e.g. const bool DEFAULT_HERO_CLOTHES_STATUS = false)